hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -494,6 +494,17 @@ private void setHmsTableParameters(
     // remove any props from HMS that are no longer present in Iceberg table props
     obsoleteProps.forEach(parameters::remove);
 
+    // altering owner
+    if (metadata.properties().get(HiveCatalog.HMS_TABLE_OWNER) != null) {
+      tbl.setOwner(metadata.properties().get(HiveCatalog.HMS_TABLE_OWNER));
+    }
+
+    // dropping owner: instead of leaving the owner blank/null, the owner will be
+    // default to whoever is making the current drop operation
+    if (obsoleteProps.contains(HiveCatalog.HMS_TABLE_OWNER)) {
```

Comment:
I wondering if the "drop ownership" use-case makes any sense. In case I'm not satisfied with the current owner of a table instead of dropping the owner I could assign it to someone (e.g. myself). I feel that setting the owner to the user who runs the updateProperties() query seems more like a side-effect rather than what a user might expect.
However, I'm open for other opinions here.

Reply:
Also too don't know of the use-case, but makes sense to me to set it to null here.

Reply:
@gaborkaszab @szehon-ho I realized something: if we created table with default owner, then we won't be able to drop the table owner (flip from the default owner/current user to null). Reason is when creating a table without owner specified, the owner property is not tracked by Iceberg Metadata file (even though there is an owner - the default owner). The later drop will just be a noop instead of setting owner to null since it cannot detect this owner property disappeared.

So I guess we have two way forward:
1. We can totally ignore drop owner, so dropping owner of all table becomes a noop
2. We can force creating table to always write the owner name to Iceberg Metadata file, whether with owner specified or not (when not specified then the default owner)

In anyway, the current implementation even after switched to set null is wrong, it causes inconsistent behavior:
1. When invoking drop owner on a table created with owner specified, it will set null
2. When invoking drop owner on a table created without owner specified, it will be a noop

Reply:
I like option 2, I think metadata file should be consistent with HMS metadata.  @gaborkaszab what do you think?

Reply:
@haizhou-zhao, sorry for the delay with my answer.

I might miss something here but I don't think that table ownership should be present in the Iceberg Metadata files nor should be a table property, it's rather an attribute of the table in HMS that could be provided during table creation. If you check [this line](https://github.com/apache/iceberg/blob/333227fbd13821365cec1bdbfcb9314a239bea0f/hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java#L374) I deliberately remove the HMS_TABLE_OWNER from the table properties because this information should only present for tables in HiveCatalog in HMS. The reason TableMetadata.properties() is used for setting the table ownership is that there was no other way to pass this piece of information to HMS without changing the Iceberg API. But HMS_TABLE_OWNER is eventually not added as an actual table param.

Following this logic I don't think that removing table ownership makes sense as a table should always have an owner. I'd vote for rejecting such requests rather than setting some default value. If the current user wants to grab the ownership of the table then it can change the ownership rather than removing it.
So in general how I imagine this is to allow changing ownership via setting a table property (but at the end this shouldn't end up as an actual table property) and I'd not allow the user to remove table ownership.

Reply:
@gaborkaszab First of all, at this moment, the actual logic of creating table differs from what you think, because all properties, including ownership, gets into iceberg metadata file on creation: [ref](https://github.com/apache/iceberg/blob/333227fbd13821365cec1bdbfcb9314a239bea0f/core/src/main/java/org/apache/iceberg/BaseMetastoreCatalog.java#L191)

Secondly, I have a different thought on whether ownership should be in iceberg metadata file or not. Iceberg supports many forms of metastores and today Iceberg users are allowed to migrate their tables between metastores, such as between Hive Metastore (HMS), Nessie, JDBC, Dynamo, etc. [ref](https://github.com/apache/iceberg/blob/32cfb62fdb9dca215250e4ca8285754e9393767d/core/src/main/java/org/apache/iceberg/BaseMetastoreCatalog.java#L69) So I think the question becomes, whether Iceberg users only care about ownership when they are using HMS, or they they also care when they use other forms of metastore. I tend to believe that ownership is an Iceberg table attributes, not specific to any forms of metastore. Otherwise, a user who wants to migrate a table from Hive to Nessie, for example, would find the ownership info lost over the process of migration (and they have to manually set the ownership again in Nessie).

Lastly, I favor allowing user to drop ownership, but I'm good with following your thought there to not allow. I can't imagine a use case for dropping ownership (yes, I agree with you why not just alter). Yet, I see no harm in allowing users to drop ownership, and I see ownership as a normal property just like others - if we are allowed to drop other iceberg or metastore properties, then making a logical exception for the ownership property so that the "unset" (set null/clear) operation is forbidden might be some inconsistency that users need to remember and be aware of when they use the code.

Reply:
Thanks for the heads-up, @haizhou-zhao!

I did some debugging and HMS_TABLE_OWNER indeed makes it into the TableMetadata not just to HMS. However, when I introduced this property my intention was to use it only for table creation to pass this to HMS and not to make it part of the table properties nor the table metadata. This is the reason I explicitly remove it from the table properties [here](https://github.com/apache/iceberg/blob/eeb055a37705b8c8ff1e13acefe0ba24382c66fe/hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java#L374). The reason this had to be provided as a table property is that this approach didn't require any Iceberg API changes, and the purpose was to pass it as table property, give it to HMS during table creation and then drop from the table properties as it have never existed there. Unfortunately as I see, as a side effect it made it into table metadata, that is unintentional (nice catch, btw).

So long story short, I don't think that any of the Iceberg clients use HMS_TABLE_OWNER as a table metadata at all. I'm certain that it is not used for determining the owner of the table (I'm not sure how other catalogs keep track of this info, if they do at all).

So that's why I was confused why we want to keep the owner in the metadata sync with the one in HMS because I only intended to add this piece of info into HMS and wasn't aware that accidentally I added it to the metadata as well.
I'm wondering if it makes sense to get rid of HMS_TABLE_OWNER from the table metadata? It would make altering the owner difficult, right?

A remark on the "drop ownership" part: well, it might be possible to drop other properties, but I still feel that dropping ownership simply doesn't make any sense especially that after dropping we have to set something as an owner and regardless of whoever we set, it feels like a side effect. I'd vote for not allow dropping ownership.

Reply:
Thanks @gaborkaszab.

First, I think altering and dropping ownership are both doable even if not having owner property in iceberg Metadata. Maybe more if branches, but ultimately should be achievable.

Summarizing our discussion above, the core question is: whether or not "owner" is an iceberg table/namespace property.

@gaborkaszab I think you are arguing that the answer should be no, and I agree that all your points make perfect sense when the answer is no (that the "owner" property should have no business with Iceberg Metadata and should only be tracked by the metastore/catalog service people uses, whether it's Hive, Nessie, DynamoDB or JDBC)

@szehon-ho and @danielcweeks would love to hear your opinion and vote before committing further.